src/interdiff.c

```diff
@@ -58,6 +59,25 @@
 #define PATCH "patch"
 #endif
 
+/* ANSI color codes for diff output */
+#define COLOR_RESET     "\033[0m"
+#define COLOR_HEADER    "\033[1m"       /* Bold for headers */
+#define COLOR_FILE      "\033[1m"       /* Bold for filenames */
+#define COLOR_HUNK      "\033[36m"      /* Cyan for hunk headers */
+#define COLOR_ADDED     "\033[32m"      /* Green for added lines */
+#define COLOR_REMOVED   "\033[31m"      /* Red for removed lines */
+#define COLOR_CONTEXT   ""              /* No color for context lines */
```

Comment:
You can create an array of these strings and index them based on `enum line_type` to eliminate the big switch statement in print_color(). No need to keep the color code macros around, like this:
```
static const char *const color_of_line[LINE_TYPE_MAX] = {
	[LINE_ADDED] = "\033[32m",	/* Green for added lines */
	[LINE_REMOVED] = "\033[31m",	/* Red for removed lines */
	...
};
```